buyer-api/src/queues/retryFailed.js

```diff
@@ -0,0 +1,52 @@
+import { web3, logger } from '../utils';
+import { getTransactionReceipt } from '../facades/helpers';
+import { dataResponseQueue } from './dataResponseQueue';
+
+const retryPendingDataResponse = async (job) => {
+  const { name, data: { orderAddress, sellerAddress, receipt } } = job;
+
+  const jobName = name ===
+    'closeDataResponseSent' ? 'closeDataResponse' : 'buyData';
```

Comment:
I just realized that the case for `buyData` is a little bit more complex: If the seller has already been accepted, the `addDataResponse` will not enqueue a job for the `closeDataResponse` since we are now doing it through the `DataAdded` event.
I would suggest changing this to:
```js
const sellerAccepted = await dataOrder.methods.hasSellerBeenAccepted(seller).call();
const jobName = sellerAccepted ? 'closeDataResponse' : 'buyData';
```
With this approach you can go directly to enqueue the job without calling `getTransactionReceipt`

Reply:
Nice catch!